x/rollapp/keeper/msg_server_app.go

```diff
@@ -0,0 +1,110 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/dymensionxyz/dymension/v3/x/rollapp/types"
+)
+
+// AddApp adds a new app
+func (k msgServer) AddApp(goCtx context.Context, msg *types.MsgAddApp) (*types.MsgAddAppResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := k.checkInputs(ctx, msg); err != nil {
+		return nil, err
+	}
+
+	// signal to client that ordering is up to them
+	if msg.Order == 0 {
+		msg.Order = -1
+	}
+
+	// charge the app creation fee
+	creator := sdk.MustAccAddressFromBech32(msg.Creator)
+	appCost := sdk.NewCoins(k.AppCreationCost(ctx))
+
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(ctx, creator, types.ModuleName, appCost); err != nil {
+		return nil, types.ErrAppCreationCostPayment
+	}
+
+	if err := k.bankKeeper.BurnCoins(ctx, types.ModuleName, appCost); err != nil {
+		return nil, types.ErrAppCreationCostPayment
+	}
+
+	app := msg.GetApp()
+	k.SetApp(ctx, app)
+
+	if err := ctx.EventManager().EmitTypedEvent(app.GetAddedEvent()); err != nil {
+		return nil, err
+	}
+
+	return &types.MsgAddAppResponse{}, nil
+}
+
+// UpdateApp updates an existing app
+func (k msgServer) UpdateApp(goCtx context.Context, msg *types.MsgUpdateApp) (*types.MsgUpdateAppResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if err := k.checkInputs(ctx, msg); err != nil {
```

Comment:
check inputs check if the app already exists and if so returns an error. 
currently the update case will always fail, as we check now if the app already exists in the check inputs, and if so fail (which obviously it will exist if it's an update operation)
do we have any tests for it?